File: src/entities/campo_template.py
from enum import Enum
from dataclasses import dataclass

class TipoDado(Enum):
    TEXTO = 'TEXTO'
    NUMERO_INTEIRO = 'NUMERO_INTEIRO'
    NUMERO_DECIMAL = 'NUMERO_DECIMAL'
    DATA = 'DATA'
    LOGICO = 'LOGICO'
    VAZIO = 'VAZIO'

@dataclass
class CampoTemplate:
    nome_campo: str
    tipo_dado: TipoDado

@dataclass
class CampoTemplateResponse:
    id_campo_template: int
    nome_campo: str
    tipo_dado: TipoDado

File: src/app/template/campo-template.ts
export enum TipoDado{
    TEXTO = 'TEXTO',
    NUMERO_INTEIRO = 'NUMERO_INTEIRO',
    NUMERO_DECIMAL = 'NUMERO_DECIMAL',
    DATA = 'DATA',
    LOGICO = 'LOGICO',
}

export class CampoTemplate{
    id_campo_template?: number;
    nome_campo? : string;
    tipo_dado?: TipoDado;

    constructor(nomeCampo: string, tipoDado: TipoDado){
        this.nome_campo = nomeCampo;
        this.tipo_dado = tipoDado;
    }
}

export class CampoTemplateRequest{
    id_campo_template?: number;
    nome_campo? : string;
    tipo_dado?: TipoDado;

    constructor(campo_template: CampoTemplateRequest){
        this.id_campo_template = campo_template.id_campo_template;
        this.nome_campo = campo_template.nome_campo;
        this.tipo_dado = campo_template.tipo_dado;
    }
}


